packages/foam-vscode/src/features/utility-commands.ts

```diff
@@ -1,54 +1,72 @@
 import * as vscode from 'vscode';
 import { FoamFeature } from '../types';
 import { URI } from '../core/model/uri';
-import { fromVsCodeUri, toVsCodeUri } from '../utils/vsc-utils';
+import { fromVsCodeUri, toVsCodeRange, toVsCodeUri } from '../utils/vsc-utils';
 import { NoteFactory } from '../services/templates';
+import { Foam } from '../core/model/foam';
 
 export const OPEN_COMMAND = {
   command: 'foam-vscode.open-resource',
   title: 'Foam: Open Resource',
 
-  execute: async (params: { uri: URI }) => {
-    const { uri } = params;
-    switch (uri.scheme) {
-      case 'file':
-        return vscode.commands.executeCommand('vscode.open', toVsCodeUri(uri));
-
-      case 'placeholder':
-        const title = uri.path.split('/').slice(-1)[0];
-
-        const basedir =
-          vscode.workspace.workspaceFolders.length > 0
-            ? fromVsCodeUri(vscode.workspace.workspaceFolders[0].uri)
-            : fromVsCodeUri(vscode.window.activeTextEditor?.document.uri)
-            ? URI.getDir(
-                fromVsCodeUri(vscode.window.activeTextEditor!.document.uri)
-              )
-            : undefined;
-
-        if (basedir === undefined) {
-          return;
-        }
-
-        const target = URI.createResourceUriFromPlaceholder(basedir, uri);
-
-        await NoteFactory.createForPlaceholderWikilink(title, target);
-        return;
-    }
-  },
-
   asURI: (uri: URI) =>
     vscode.Uri.parse(`command:${OPEN_COMMAND.command}`).with({
       query: encodeURIComponent(JSON.stringify({ uri: URI.create(uri) })),
     }),
 };
 
 const feature: FoamFeature = {
-  activate: (context: vscode.ExtensionContext) => {
+  activate: (context: vscode.ExtensionContext, foamPromise: Promise<Foam>) => {
     context.subscriptions.push(
       vscode.commands.registerCommand(
         OPEN_COMMAND.command,
-        OPEN_COMMAND.execute
+        async (params: { uri: URI }) => {
```

Comment:
moved the `execute` here as we need `foam` here and trying to be smart about initialization was more convoluted than just doing this.

Commands need a bit of love/refactoring, but this is out of scope for this PR